### DISCUSSION TITLE
DrawString make text flicker with FreeFont

### DISCUSSION THREAD

**Discussion by meilleur102** [Bodmer/TFT_eSPI#2535] [Q&A - General] [answered]
Hi. I'm trying to center my text on the screen and I found a solution here on the forum
  
```
tft.setTextDatum(MC_DATUM);
tft.drawString(data, tft.width() / 2, tft.height() / 2);
```

But when I use a FreeFont the text flicker. 
If I use no font (the OEM one) everything is ok. 

Thanks
Jc

**Comment by Bodmer** [comment, marked as answer]
The only way around this is to render the font in a sprite and then push on each update. Smooth fonts should render with less flicker.

**Comment by meilleur102** [comment]
ok thanks. But the problem I have is if I want to have the hole screen I don't have enought ram in the ESP32. 
I have a 480x320 screen. Is it possible to put the Sprite on the SD card or another way to have 480x320 sprite?

**Reply by Bodmer** [reply]
The sprite has to be in RAM. If your board has additional PSRAM then full screen sprites are possible. Alternatively if the screen is divided into zones, then the sprite can  be the zone size and reused for each zone.

This library is another option:
https://github.com/Bodmer/FlickerFreePrint

**Reply by meilleur102** [reply]
Ok and is there a way to see the free ram in a board? 
Maybe I don't have the good board for the application I want to make. 
I want to have a Dashboard of a F1 sim Racing steering wheel. 
There's a lot of information, but it's only text. 

I have a GIF but it's only there at the start of the program and after that I fill the screen with information. 
IDK if I need to "delete" the gif or if's it's better to put it on a SD card.

The information change quickly on the steering so I need a fast board. This is why first I was thinking about the ESP32. 
The "lap time" is display in milliseconds and with the ESP there's no lag when I display with the OEM font. 

Ill check the FlickerFreePrint maybe it will work. 
If not , can you recommend a board that will be fast enough and have enough ram? 

Thanks a lot for your help.
Amazing Library btw.

EDIT:
The FlickerFree seems to do the tricks! Thanks

**Comment by TinyMega** [comment]
For smooth fonts you need to pass an extra parameter to setTextColor() in order to include background updates.
